### PR TITLE
Add Foreman client mash and repoclosure to Foreman nightly

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

**Comment by ehelms** [comment]
@ekohl I debated internally whether to make the client repositories their own pipeline independent of the core push. While not currently present, the aspects I wondered about were adding tests of client packages and if that should be independent of the release pipeline for core.

**Comment by ekohl** [comment]
I think it'll depend on if we move hammer over as well but I think that does make sense. Client repos should be usable without the core repository so should be promoteable independently.

**Comment by ehelms** [comment]
I've updated this to be it's own client release job.

**Comment by ehelms** [comment]
This will need https://github.com/theforeman/foreman-packaging/pull/2821 before it can hope to succeed

File: puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/foremanClientRelease.groovy [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+            }
+        }
+    }
+}
+
+void repoclosure(repo, dist) {
+    obal(
+        action: 'repoclosure',
+        packages: "${repo}-repoclosure-${dist}"
+    )
```
Comment (ehelms) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/foremanClientRelease.groovy]: Switching to try out using our obal repoclosure config and functionality.

**Pull request merged**